editor/editor_help.cpp

```diff
@@ -43,8 +44,53 @@
 
 #define CONTRIBUTE_URL vformat("%s/contributing/documentation/updating_the_class_reference.html", VERSION_DOCS_URL)
 
+// TODO: this is sometimes used directly as doc->something, other times as EditorHelp::get_doc_data(); Might this be a problem?
 DocTools *EditorHelp::doc = nullptr;
```

Comment:
I guess this is done on purpose? Using the method ensures that the thread used for help generation has finished, but accessing it directly, obviously, doesn't. So I think that if the help generation does it, then it's okay, but if something else does it, it's probably not okay.

For explicitness' sake I would avoid accessing it directly and add a `_get_doc_data_unsafe()` method, or something like that, for places that need access without checking for the thread,

Reply:
But this can probably done in a follow-up, as if it's a problem, it's pre-existing.

Reply:
Yeah, I've not really touched on any threading stuff with Godot so I'm not 100% certain whether there is even a problem in the first place. Do you think it's appropriate to leave the comment there?

Reply:
Yes, it's good to have a comment here pointing at a potential issue.